generators/typescript-v2/ast/src/ast/Type.ts

```diff
@@ -2,12 +2,99 @@ import { assertNever } from "@fern-api/core-utils";
 import { AstNode } from "./core/AstNode";
 import { Writer } from "./core/Writer";
 
-type InternalType = String_;
+type InternalType =
+    | String_
+    | Number_
+    | BigInt_
+    | Boolean_
+    | Array_
+    | Object_
+    | Map_
+    | Tuple
+    | Enum
+    | Any
+    | Unknown
+    | Void
+    | Undefined
+    | Null
+    | Never
+    | Nop;
```

Comment:
What's `Nop`? Is it supposed to be `NoOp` or `Nope`?

Reply:
It's no-op. In the go-v2 generator, we use the same abbreviation so I'm keeping it as nop for now.

Reply:
This is the node I used in [the other AST](https://github.com/fern-api/fern/blob/0c3d46efacba93049a6485cce44394955c0eac5d/generators/go-v2/ast/src/ast/TypeInstantiation.ts#L22) - it's a word I've used in the past for `no-op` since it's always weird to see `Noop` or `NoOp`. We can change this later though.